rotkehlchen/accounting/history_base_entries.py

```diff
@@ -0,0 +1,257 @@
+    result[deposit_key] = HistoryBaseEntrySettings(
+        taxable=False,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='spend',
+    )
+    withdraw_key = str(HistoryEventType.WITHDRAWAL) + '__' + str(HistoryEventSubType.NONE)
+    result[withdraw_key] = HistoryBaseEntrySettings(
+        taxable=False,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    fee_key = str(HistoryEventType.SPEND) + '__' + str(HistoryEventSubType.FEE)
+    result[fee_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        count_entire_amount_spend=True,
+        count_cost_basis_pnl=True,
+        method='spend',
+    )
+    renew_key = str(HistoryEventType.RENEW) + '__' + str(HistoryEventSubType.NONE)
+    result[renew_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        count_entire_amount_spend=True,
+        count_cost_basis_pnl=True,
+        method='spend',
+    )
+    swap_key = str(HistoryEventType.TRADE) + '__' + str(HistoryEventSubType.SPEND)
+    result[swap_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=True,
+        method='spend',
+        accounting_treatment=TxAccountingTreatment.SWAP,
+    )
+    airdrop_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.AIRDROP)
+    result[airdrop_key] = HistoryBaseEntrySettings(
+        taxable=LedgerActionType.AIRDROP in pot.settings.taxable_ledger_actions,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    reward_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.REWARD)
+    result[reward_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    return result
+
+
+def history_base_entries_iterator(
+        events_iterator: Iterator[AccountingEventMixin],
+        associated_event: HistoryBaseEntry,
+) -> Iterator[HistoryBaseEntry]:
+    """
+    Takes an iterator of accounting events and transforms it into a history base entries iterator.
+    Takes associated event as an argument to be able to log it in case of errors.
+    """
+    for event in events_iterator:
+        if not isinstance(event, HistoryBaseEntry):
+            log.error(
+                f'At accounting for event {associated_event.notes} with identifier '
+                f'{associated_event.event_identifier} we expected to take an additional '
+                f'event but found a non history base entry event',
+            )
+            return
+
+        yield event
+
+
+class HistoryBaseEntriesAccountant():
```

Comment:
nitpick: in python3 is not needed

```suggestion
class HistoryBaseEntriesAccountant:
```